### PR TITLE
Sync Input

### LINKED ISSUE #41: [Syncing] Input
## Points to note
- Component in action - `Input.js`
- Github Urls to this component:
  * [neeto-ui/Input.js](https://github.com/bigbinary/neeto-ui/tree/67ee4a5188c5feccbeca2767107651fb24932764/lib/components/Input.js)
  * [NitroUI/Input.js](https://github.com/bigbinary/neeto-help-web/tree/0b11a7b0f98ac7c533df1e9ec37cf900b4cbc901/app/javascript/Admin/Components/NitroUI/Input.js)
- :green_circle: Green color signifies how that particular line is written in `neeto-ui` package
- :red_circle: Red color signifies how that particular line is written in `NitroUI` internals

## Sync Required or Not
- Are there significant changes for `Input.js` that needs to be synced to `neeto-ui` package from `NitroUI`? - **No**
- `nakedInput` is named as `naked` in neetoui. But i'm not sure what exactly it's used for. What exactly does prefixing with `naked` mean from `className` perspective? 

## Diff
```diff
--- NitroUI/Input.js	2020-11-28 16:54:23.974302100 +0530
+++ neetoui-official/Input.js	2020-11-28 16:49:18.736973600 +0530
@@ -1,10 +1,10 @@
-import React from "react";
+import React, { Component } from "react";
 import classnames from "classnames";
 import Label from "./Label";
 
-const Input = React.forwardRef(
-  (
-    {
+class Input extends Component {
+  render() {
+    const {
       size,
       type,
       label,
@@ -16,21 +16,20 @@
       helpText,
       onChange,
       className,
+      inputClassName,
+      naked,
       maxLength,
       labelProps,
-      nakedInput,
       dataTestId,
-      contentSize,
       placeholder,
       leadingIcon,
       required = false,
       hideCharacterCount,
       ...otherProps
-    },
-    ref
-  ) => {
+    } = this.props;
+
     const baseClasses =
-      "form-input block w-full text-sm flex flex-row justify-start items-center transition-all duration-200";
+      "form-input block w-full sm:text-sm sm:leading-5 flex flex-row justify-start items-center transition-all duration-200";
     const errorClasses =
       "border-red-400 text-red-800 placeholder-red-300 focus-within:shadow-focus-red";
 
@@ -44,7 +43,7 @@
     return (
       <div
         className={classnames([
-          "flex flex-col items-start justify-start flex-grow",
+          "flex flex-col items-start justify-start",
           className
         ])}
       >
@@ -62,11 +61,7 @@
             {label}
           </Label>
         )}
-        <div
-          className={classnames(["relative w-full rounded-md"], {
-            "shadow-sm": !nakedInput
-          })}
-        >
+        <div className={classnames("relative w-full rounded-md", { "shadow-sm": !naked, "border-none": naked })}>
           {leadingIcon && (
             <div className="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none">
               <i className={leadingIcon}></i>
@@ -74,42 +69,39 @@
           )}
           <div
             className={classnames(baseClasses, {
-              "focus-within:shadow-focus-purple focus-within:border-purple-400":
-                !inputError && !nakedInput,
-              [errorClasses]: !!inputError,
+              "focus-within:shadow-focus-purple focus-within:border-purple-400": !inputError && !naked,
+              [errorClasses]: !!inputError && !naked,
+              "border-none": naked,
               "bg-gray-50 cursor-not-allowed": disabled,
               "pr-10": !suffix && !!inputError,
               "pl-10": leadingIcon,
-              "py-1.5": size === "small",
-              "border-none px-0": nakedInput
+              "py-1.5": size === "small"
             })}
           >
             {prefix && (
               <div className="flex items-center pr-1">
-                <span className="flex flex-row items-center justify-start text-gray-500">
+                <span className="text-gray-500 sm:text-sm sm:leading-5">
                   {prefix}
                 </span>
               </div>
             )}
             <input
-              ref={ref}
               onChange={onChange}
-              value={value ?? ""}
+              value={value || ""}
               type={type}
               placeholder={placeholder}
               className={classnames(
-                ["w-full focus:outline-none bg-transparent"],
+                ["w-full focus:outline-none bg-transparent", inputClassName],
                 { "cursor-not-allowed": disabled }
               )}
               disabled={disabled}
               data-test-id={dataTestId}
               maxLength={maxLength}
-              size={contentSize}
               {...otherProps}
             />
             {suffix && (
               <div className="flex items-center pl-1">
-                <span className="flex flex-row items-center justify-start text-gray-500">
+                <span className="text-gray-500 sm:text-sm sm:leading-5">
                   {suffix}
                 </span>
               </div>
@@ -123,7 +115,7 @@
       </div>
     );
   }
-);
+}
 
 const ErrorSVG = () => (
   <div className="absolute inset-y-0 right-0 flex items-center pr-3 pointer-events-none fadeIn">
```

### PR DESCRIPTION
**Pull request by yedhink**
Fixes #41 

@edwinbbu _a
Please review and merge if everything looks good.

cc: @goutham-subramanyam

### CODE REVIEW

File: lib/components/Input.js [hunk 1]
```diff
@@ -1,121 +1,127 @@
...
+    prefix,
+    disabled,
+    helpText,
+    onChange,
+    className,
+    inputClassName,
+    maxLength,
+    labelProps,
+    nakedInput,
+    dataTestId,
```
Comment (goutham-subramanyam) [inline lib/components/Input.js:22]: We should be removing this prop from all our components. We should be passing it as `data-test-id` itself and pass it along as `otherProps`

**Review by goutham-subramanyam — CHANGES_REQUESTED** [review]
@yedhink _a

File: lib/components/Input.js (hunk 1 above)
Comment (yedhink) [reply lib/components/Input.js:22]: @goutham-subramanyam 

Will handle this is in a separate issue if it needs to be removed from all components. Or are you suggesting to make that change in this PR itself? 

If in this PR itself, then there will be a lot of changes across many components.

Awaiting your response.

**Comment by yedhink** [comment]
@goutham-subramanyam _a

Awaiting response for https://github.com/bigbinary/neeto-ui/pull/70#discussion_r537554641

File: lib/components/Input.js (hunk 1 above)
Comment (goutham-subramanyam) [reply lib/components/Input.js:22]: You can remove dataTestID for Input in this PR and handle the rest in separate issue.
@yedhink _a

**Comment by yedhink** [comment]
@goutham-subramanyam _a

I have made the requested changes. Please review.

**Comment by goutham-subramanyam** [comment]
@yedhink _a Please add default values for the props like you had done for all the other components.

**Comment by yedhink** [comment]
@goutham-subramanyam _a

Have updated. Please review.

File: lib/components/Input.js [hunk 2]
```diff
@@ -1,121 +1,125 @@
...
-      placeholder,
-      leadingIcon,
-      required = false,
-      hideCharacterCount,
-      ...otherProps
-    } = this.props;
+const Input = React.forwardRef((props, ref) => {
+  const {
+    size = "large",
+    type,
```
Comment (goutham-subramanyam) [inline lib/components/Input.js:8]:
```suggestion
    type = "text",
```

**Review by goutham-subramanyam — APPROVED** [review]
LGTM

**Pull request merged**